imap_processing/tests/glows/test_glows_l1a_data.py

```diff
@@ -361,3 +391,120 @@ def test_generate_status_data():
+        )
+        assert (
+            de.status_data.imap_sclk_next_pps
+            == validation_data["imap_sclk_next_pps"][index]
+        )
+        assert (
+            de.status_data.catbed_heater_active
+            == validation_data["catbed_heater_active"][index]
+        )
+        assert (
+            de.status_data.spin_period_valid
+            == validation_data["spin_period_valid"][index]
+        )
+        assert (
+            de.status_data.spin_phase_at_next_pps_valid
+            == validation_data["spin_phase_at_next_pps_valid"][index]
+        )
+        assert (
+            de.status_data.spin_period_source
+            == validation_data["spin_period_source"][index]
+        )
+        assert de.status_data.spin_period == validation_data["spin_period"][index]
+        assert (
+            de.status_data.spin_phase_at_next_pps
+            == validation_data["spin_phase_at_next_pps"][index]
+        )
+        assert (
+            de.status_data.number_of_completed_spins
+            == validation_data["number_of_completed_spins"][index]
+        )
+        assert (
+            de.status_data.filter_temperature
+            == validation_data["filter_temperature"][index]
+        )
+        assert de.status_data.hv_voltage == validation_data["hv_voltage"][index]
+        assert (
+            de.status_data.glows_time_on_pps_valid
+            == validation_data["glows_time_on_pps_valid"][index]
+        )
+        assert (
+            de.status_data.time_status_valid
+            == validation_data["time_status_valid"][index]
+        )
+        assert (
+            de.status_data.housekeeping_valid
+            == validation_data["housekeeping_valid"][index]
+        )
+        assert (
+            de.status_data.is_pps_autogenerated
+            == validation_data["is_pps_autogenerated"][index]
+        )
+        assert (
+            de.status_data.hv_test_in_progress
+            == validation_data["hv_test_in_progress"][index]
+        )
+        assert (
+            de.status_data.pulse_test_in_progress
+            == validation_data["pulse_test_in_progress"][index]
+        )
+        assert (
+            de.status_data.memory_error_detected
+            == validation_data["memory_error_detected"][index]
+        )
+
+        assert de.l0.LEN == validation_data["number_of_de_packets"][index]
+
+        validation_de = ast.literal_eval(validation_data["de_data"][index])
+
+        assert (
+            de.direct_events[
+                validation_data["de_data_counter"][index]
+            ].timestamp.seconds
+            == validation_de[0]
+        )
+        assert (
+            de.direct_events[
+                validation_data["de_data_counter"][index]
+            ].timestamp.subseconds
+            == validation_de[1]
+        )
+        assert (
+            de.direct_events[validation_data["de_data_counter"][index]].impulse_length
```

Comment:
Is this where the 4 is coming from earlier that I commented on? Rather than your Dataset coordinates being `np.arange(4)`, could you make it `["timestamp seconds", "timestamp subseconds", "impulse length", "multi event"]`? Storing the strings as the coordinate variable would make the enum more explicit and easier to follow I think.

Reply:
I don't think I know your prior comment, but yes, I can update the dataset coordinates. I didn't realize they could be strings!

Reply:
That actually appears to crash cdflib with an error that I have been discussing with Bryan. So I'm going to leave it as is and make a comment, and maybe go back if that error is a bug

Reply:
:( sigh, I really dislike CDF, netCDF would make this much easier.

Reply:
On the bright side, using CDF files builds character!

Reply:
Coordinates or data could be strings, but we need to use make sure data type of CDF attrs are setup properly. You could use `StringAttrs` class from global_attrs.py